unyt/tests/test_array_functions.py

```diff
@@ -663,6 +663,36 @@ def test_histogram_implicit_units():
     assert bins.units == arr.units
 
 
+def test_histogram_with_weights_density():
```

Comment:
This looks like 4 separate tests to me. Could you split them ? feel free to regroup them under a test class if you like that style.